near-plugins-derive/src/access_controllable.rs

```diff
@@ -194,6 +194,16 @@ pub fn access_controllable(attrs: TokenStream, item: TokenStream) -> TokenStream
                 Some(self.revoke_super_admin_unchecked(account_id))
             }
 
+            fn transfer_super_admin(&mut self, account_id: &::near_sdk::AccountId) -> Option<bool> {
+                let current_super_admin = ::near_sdk::env::predecessor_account_id();
+                if !self.is_super_admin(&current_super_admin) {
+                    return None;
+                }
+
+                self.revoke_super_admin_unchecked(&current_super_admin);
+                Some(self.add_super_admin_unchecked(&account_id))
+            }
```

Comment:
I suggest that we first add the super admin, and only if this is successful, then revoke the super admin. Generally, it's a better approach, especially if something will eventually be changed in these two functions on how they work. This is required to make sure the contract doesn't accidentally become locked on ACL.

CC @karim-en @mooori

Reply:
The state modifications in `acl_transfer_super_admin` should be atomic since no promises or cross contract calls are scheduled. So I think currently it’s not possible to lock Acl in that way. I assume it’s unlikely that the default `AccessControllable` implementation will schedule promises or xcc in the future. Still I see the point of making this as future proof as possible. I can prepare a PR to reorder it tomorrow.

Reply:
#101 reorders the operations.